### ISSUE TITLE
ZIO Test: Warn if non-object extends DefaultRunnableSpec

### ISSUE THREAD

**Issue by evis** [zio/zio#2696]
If we make a mistake and use a class instead of an object to create a test, there is no message about this from `sbt test`.

e.g.:

```scala
import zio.test.DefaultRunnableSpec

class MySpec extends DefaultRunnableSpec(???)
```

```
> test
[info] Done
[success] Total time: 0 s, completed Jan 23, 2020 11:20:04 PM
```

It would be nice to provide a message like this:

```
[warn] MySpec is ignored because it's not an object
```

I spent 15 minutes today trying to figure out, why my test which should fail doesn't invoke at all. :(

If we replace `class` with `object`, then the test fails as expected.

**Comment by evis** [comment]
Maybe, another option is to simply run such tests as is.

**Comment by adamgfraser** [comment]
Yes, that would be a nice feature. I don't think we can prevent classes from inheriting from `DefaultRunnableSpec` so I think we would need to do something in the SBT test runner to detect those situations and display a warning. Not sure off the top of my head what ev

**Comment by jdegoes** [comment]
Error on `class` creation.

**Comment by hearnadam** [comment]
I think we should add a check for this.

**Comment by dancewithheart** [comment]
PR #10618 addressing this for `ZIOSpecDefault` (because `DefaultRunnableSpec` no longer exists):
Class-based specs are now reported with a clear error instead of being silently ignored.

Happy to adapt if you'd prefer a different behavior (e.g. non-fatal reporting).
